### PR TITLE
[export] Prototype on serialization schema.

### PR DESCRIPTION
**Pull request by zhxchen17**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/_export/serde/schema.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+@dataclass
+class NamedArgument:
+    name: str
+    arg: Argument
+
+
+@dataclass
+class Node:
+    target: str
```
Comment (larryliu0820) [inline torch/_export/serde/schema.py:108]: hmm what is this string? Are we serializing `torch._ops.OpOverload` into a string?

Comment (zhxchen17) [reply torch/_export/serde/schema.py:108]: yeah, I think we need some have some namespace to look up operators when resolving these names.

Comment (larryliu0820) [reply torch/_export/serde/schema.py:108]: Might be useful to introduce `class Operator`? There we can add version info.

Comment (zhxchen17) [reply torch/_export/serde/schema.py:108]: yeah, I think we can do that. one side question: does every op has version?

Comment (larryliu0820) [reply torch/_export/serde/schema.py]: I think it can be optional? Do we support optional in this schema?

Comment (zhxchen17) [reply torch/_export/serde/schema.py]: yep, let's do that

File: torch/_export/serde/schema.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    backwardSignature: Optional[BackwardSignature]
+
+
+@dataclass
+class GraphModule:
+    graph: Graph
+    buffers: Dict[str, TensorMeta]
+    parameters: Dict[str, TensorMeta]
+    metadata: Dict[str, str]
+    signature: GraphSignature
```
Comment (angelayi) [inline torch/_export/serde/schema.py:157]: Also constraints for ShapeEnv? And not sure if you also want to add ShapeEnv's symbol to source in this diff too.

File: torch/_export/serde/schema.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+@dataclass
+class BackwardSignature:
+    gradientsToParameters: Dict[str, str]
+    gradientsToUserInputs: Dict[str, str]
+    lossOutput: str
+
+
+@dataclass
+class GraphSignature:
```
Comment (angelayi) [inline torch/_export/serde/schema.py:142]: Should graph signature also include the pytree in/out?

File: torch/_export/serde/schema.py (hunk 2 above)
Comment (zhxchen17) [reply torch/_export/serde/schema.py:157]: I want to leave out ShapeEnv for now, we should have some test for dynamic shape model first and then start from there.

File: torch/_export/serde/schema.py (hunk 3 above)
Comment (zhxchen17) [reply torch/_export/serde/schema.py:142]: yeah, after some thoughts I think we should contain them.

File: torch/_export/serde/schema.py (hunk 2 above)
Comment (angelayi) [reply torch/_export/serde/schema.py:159]: ok! I think rest of this looks good to me ^_^

**Review by angelayi — APPROVED** [review]

**Pull request closed without merging**